docs/technologies/vue.md

```diff
@@ -0,0 +1,183 @@
+
+The point of having a file like this to provide you with a place where you can write your test.
+
+**For Vue.js labs, you can continue with the 'Custom Template' selected by default:**
+
+![](/images/vue/lab-edit-test-file.png)
+
+Paste the following code in the editor as a starting point:
+
+```jsx
+import { mount } from '@vue/test-utils'
+
+test('Test 1', async () => {
+    try {
+        const { default: HelloWorld } = await import('../code/src/components/HelloWorld.vue')
+
+        const wrapper = mount(HelloWorld)
+
+        expect(wrapper.text()).toContain('Hello Vue.js!')
+    } catch (e) {
+        expect(false).toBe(true)
+    }
+})
+```
+
+Let us understand what is happening here exactly:
+
+-   Remember that we can code anything in this file and then execute it later. In this example, we're writing a Vue.js Vitest test script from scratch. Check out [vitest docs](https://vitest.dev) if you're new to vitest.
+-   Remember that we will install vitest and other required utilities in the evaluation script section below. Therefore, you can try to import and use anything and everything here you want.
+-   The rest of the code is just importing the default user code, and testing it through standard unit testing procedures.
+
+![](/images/html-css/playground-tests.png)
+
+-   The number of `test(...)` blocks inside your `describe` suite should match the number of challenges added in the creator area.
+
+-   **Note:** If your number of `test` blocks are less than challenges added back in the UI, the "extra" UI challenges would automatically stay as "false". If you add more challenges in test file, the results would be ignored. Therefore, it is **important** that the `results.length` is same as the number of challenges you added in the challenges UI.
+
+-   We then also add jQuery and chai for assisting with testing. Although it is not required as long as you can populate the `results` array properly.
+
+This completes your evaluation script for the lab. Your lab is now almost ready for users.
+
+## Step 5 - Evaluation Script
+
+Evaluation script is actually what runs when the user on the codedamn playground clicks on "Run Tests" button.
+
+![](/images/common/lab-run-tests.png)
+
+Remember that we're using Vue Vite playground setup. This means we can assume that we already have vite installed.
+
+However, we still need to setup a lot of things: `jsdom`, `vitest`, and `@vue/test-utils`. Therefore, we can write our evaluation bash script to install all of this and run our tests. Here's how the Vue vitest script looks like:
+
+```sh
+#!/bin/bash
+set -e 1
+
+# Assumes you are running a vue vite playground on codedamn
+
+# Install vitest and testing util
+cd /home/damner/code
+yarn add vitest@0.29.7 jsdom@21.1.1 @vue/test-utils@2.3.2 --dev
+mkdir -p /home/damner/code/__labtests
+ 
+# Move test file
+mv $TEST_FILE_NAME /home/damner/code/__labtests/all.test.js
+
+# vitest config file
+cat > /home/damner/code/__labtests/config.js << EOF
```

Comment:
Use `.labtests` folder (should be mentioned in https://teach.codedamn.com/docs/technologies/html-css as well. 

`__labtests` is deprecated now